api/scpca_portal/management/commands/dispatch_to_batch.py

```diff
@@ -62,29 +68,41 @@ def submit_job(
 
         logger.info(f'{job_name} submitted to Batch with jobId {response["jobId"]}')
 
-    def dispatch_to_batch(self, project_id: str = "", **kwargs):
+    def dispatch_to_batch(self, project_id: str, regenerate_all: bool, **kwargs):
         """
-        Iterate over all projects that don't have computed files and submit each
-        resource_id and download_config combination to the Batch queue.
-        If a project id is passed, then computed files are created for all combinations
-        within that project.
+        Iterate over all projects that fit the criteria of the passed flags
+        and submit jobs to Batch accordingly.
         """
         projects = (
             Project.objects.filter(project_computed_files__isnull=True)
-            if not project_id
-            else Project.objects.filter(scpca_id=project_id)
+            if not regenerate_all
+            else Project.objects.all()
         )
 
+        if project_id:
+            projects.filter(scpca_id=project_id)
```

Comment:
This doenst look right to me, my understanding is that querysets are immutable. I would first try to make your tests fail with this implementation then fix this to overwrite the `projects` queryset.